### PR TITLE
[CAMEL-9869] Create Apache Flink Component

### PR DESCRIPTION
**Pull request by sbcd90**
Hello @davsclaus ,

I have enhanced the `camel-flink` producer implementation to support `Flink datastream`. 
Please review.

### CODE REVIEW

File: components/camel-flink/src/main/java/org/apache/camel/component/flink/FlinkConstants.java [hunk 1]
```diff
@@ -14,15 +14,16 @@
...
  * limitations under the License.
  */
+
 package org.apache.camel.component.flink;
 
-public final class FlinkConstants {
+public class FlinkConstants {
 
-    public static final String FLINK_DATASET_HEADER = "CamelFlinkDataSet";
+    public static final String FLINK_DATASET_HEADER = "CAMEL_FLINK_DATASET";
```
Comment (davsclaus) [inline components/camel-flink/src/main/java/org/apache/camel/component/flink/FlinkConstants.java]: The standard for keys in Camel is as in that red line, eg camelComponentNameKeyName

File: components/camel-flink/src/main/java/org/apache/camel/component/flink/FlinkEndpoint.java [hunk 2]
```diff
@@ -64,9 +72,12 @@ protected void doStart() throws Exception {
     public Producer createProducer() throws Exception {
         if (endpointType == EndpointType.dataset) {
             return new DataSetFlinkProducer(this);
-        } else {
-            throw new UnsupportedOperationException("datastream not yet supported");
         }
+        else if (endpointType == EndpointType.datastream) {
```
Comment (davsclaus) [inline components/camel-flink/src/main/java/org/apache/camel/component/flink/FlinkEndpoint.java]: Run from command line this goal: mvn install -Psourcecheck in the camel-flink directory
and then fix those checkstyle issues it reports

File: components/camel-flink/src/main/java/org/apache/camel/component/flink/VoidDataSetCallback.java [hunk 3]
```diff
@@ -14,6 +14,7 @@
  * See the License for the specific language governing permissions and
  * limitations under the License.
  */
+
```
Comment (davsclaus) [inline components/camel-flink/src/main/java/org/apache/camel/component/flink/VoidDataSetCallback.java]: Do not have empty line here

**Comment by sbcd90** [comment]
Hello @davsclaus ,

Thanks for your review & comments.
I have made all the necessary changes.

- standard for keys in camel is maintained.
- all checkstyle issues are fixed.
- all empty lines are removed after the license.

Please review now.

**Pull request merged**